src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.cs

```diff
@@ -45,6 +51,9 @@ internal partial class FindReferencesSearchEngine
             _options = options;
 
             _progressTracker = progress.ProgressTracker;
+
+            var pair = new ConcurrentExclusiveSchedulerPair();
+            _scheduler = _options.Parallel ? pair.ConcurrentScheduler : pair.ExclusiveScheduler;
```

Comment:
wasn't sure if there was another way to accompliush this.  I could use the default TaskScheduler for the parallel case isntead.  Do we have an existing serial-task-scheduler somewhere i could use?

Reply:
The `Task.Run` call adds a bunch of overhead. I'd have two paths: one that uses `Task.Run` and a second which doesn't use that path at all. There is no need for task schedulers on either path.

Reply:
i started taht way, but it was very complex and unpleasant.  It also meant we'd need to maintain two sets of code that we'd have to keep in sync (no pun intended).

Reply:
I'm not particularly concerned about the overhead here because the cost of the individual find operations is *much* greater.  teh simplicity this buys is is one set of logic, that just can be easily configured to run with more/less computer resources as appropriate.

Reply:
(i literaly did start by having the two codepaths.  but it just got uglier and uglier and i couldn't bear it) :)

Reply:
works for me